test/services/onlinetesting/OnlineTestServiceSpec.scala

```diff
@@ -136,62 +150,107 @@ class OnlineTestServiceSpec extends PlaySpec with BeforeAndAfterEach with Mockit
-      doThrow(classOf[NotFoundException]).when(otRepositoryMock).getOnlineTestDetails("nonexistent-userid")
-      intercept[NotFoundException] {
-        onlineTestService.getOnlineTest("nonexistent-userid")
-      }
+    "return None if the user does not exist" in new OnlineTest {
+      when(appRepositoryMock.findCandidateByUserId(any[String])).thenReturn(Future.successful(None))
+      val result = onlineTestService.getPhase1TestProfile("nonexistent-userid").futureValue
+      result mustBe None
+
     }
 
     val validInviteDate = new DateTime(2016, 5, 21, 0, 0)
     val validExpireDate = new DateTime(2016, 6, 9, 0, 0)
 
     "return a valid set of aggregated online test data if the user id is valid" in new OnlineTest {
-      when(otRepositoryMock.getOnlineTestDetails("valid-userid")).thenReturn(Future.successful(
-        OnlineTestDetails(
-          inviteDate = validInviteDate,
-          expireDate = validExpireDate,
-          onlineTestLink = "http://www.google.co.uk",
-          cubiksEmailAddress = "test@test.com",
-          isOnlineTestEnabled = true
-        )
+      when(appRepositoryMock.findCandidateByUserId(any[String])).thenReturn(Future.successful(
+        Some(candidate)
       ))
-      when(appRepositoryMock.findCandidateByUserId("valid-userid")).thenReturn(Future.successful(
-        Some(Commands.Candidate(
-          "valid-userid",
-          Some("valid-applicationid"),
-          None,
-          Some("Bruce"),
-          Some("Wayne"),
-          Some("Batman"),
-          None,
-          None,
-          None,
-          None
+
+      when(otRepositoryMock.getPhase1TestProfile(any[String])).thenReturn(Future.successful(
+        Some(Phase1TestProfile(expirationDate = validExpireDate,
+          tests = List(phase1Test)
         ))
       ))
-      when(otprRepositoryMock.hasReport("valid-applicationid")).thenReturn(Future.successful(
-        false
-      ))
 
-      val result = onlineTestService.getOnlineTest("valid-userid").futureValue
+      val result = onlineTestService.getPhase1TestProfile("valid-userid").futureValue
 
-      result.cubiksEmailAddress must equal("test@test.com")
-      result.isOnlineTestEnabled must equal(true)
-      result.pdfReportAvailable must equal(false)
-      result.onlineTestLink must equal("http://www.google.co.uk")
-      result.expireDate must equal(validExpireDate)
-      result.inviteDate must equal(validInviteDate)
+      result.get.expirationDate must equal(validExpireDate)
+      result.get.tests.head.invitationDate must equal(InvitationDate)
     }
   }
 
   "register and invite application" should {
+
+    "issue one email for invites to SJQ for GIS candidates" in new OnlineTest {
+
+      when(cubiksGatewayClientMock.registerApplicant(eqTo(registerApplicant))(any[HeaderCarrier]))
+        .thenReturn(Future.successful(registration))
+      when(cubiksGatewayClientMock.inviteApplicant(any[InviteApplicant])(any[HeaderCarrier]))
```

Comment:
I'd probably move this mock setup into a testfixture trait at the bottom of the file, leaves the test clean and readable